apps/common/models/i-casework-getcase.js

```diff
@@ -15,10 +15,11 @@ module.exports = class DocumentModel extends Model {
     return crypto.createHash('md5').update(this.get('reference-number') + date + config.icasework.secret).digest('hex');
   }
 
-  prepare(data) {
-    const props = super.prepare(data);
-    props.CaseId = this.get('reference-number');
-    return props;
+  prepare() {
+    return super.prepare().then(props => {
```

Comment:
Because of the Compose function above and needed to fetch a token bearer, it means that prepare step has now become async. As it overrides any other prepare steps declared in other models to be on the safe side, and match the pattern in the HOF model, all declared prepare steps have been handled to be async. So here we assume super.prepare is async...